boa/src/builtins/array/mod.rs

```diff
@@ -399,6 +401,179 @@ impl Array {
+                Some(constructor) => constructor
+                    .construct(&[], this, context)?
+                    .as_object()
+                    .cloned()
+                    .ok_or_else(|| {
+                        context.construct_type_error("Object constructor didn't return an object")
+                    })?,
+                _ => Self::array_create(0, None, context)?,
+            };
+
+            // c. Let iteratorRecord be ? GetIterator(items, sync, usingIterator).
+            let iterator_record =
+                items.get_iterator(context, Some(IteratorHint::Sync), Some(using_iterator))?;
+
+            // IfAbruptCloseIterator is a shorthand for a sequence of algorithm steps that use an Iterator Record
+            macro_rules! if_abrupt_close_iterator {
+                ($val:expr) => { match $val {
+                    // 2. Else if value is a Completion Record, set value to value.
+                    Ok(val) => val,
+                    // 1. If value is an abrupt completion, return ? IteratorClose(iteratorRecord, value).
+                    Err(err) => return iterator_record.close(Err(err), context)
+                } }
+            }
+
+            // d. Let k be 0.
+            // e. Repeat,
+            //     i. If k ≥ 2^53 - 1 (MAX_SAFE_INTEGER), then
+            //     ...
+            //     x. Set k to k + 1.
+            for k in 0..9_007_199_254_740_991_u64 {
+                let next = if_abrupt_close_iterator!(iterator_record.next(context));
+
+                // iv. If next is false, then
+                if next.done {
+                    // 1. Perform ? Set(A, "length", 𝔽(k), true).
+                    a.set("length", k, true, context)?;
+                    // 2. Return A.
+                    return Ok(a.into());
+                }
+
+                // v. Let nextValue be ? IteratorValue(next).
+                let mapped_value = if let Some(mapfn) = mapping {
+                    // vi. If mapping is true, then
+                    //     1. Let mappedValue be Call(mapfn, thisArg, « nextValue, 𝔽(k) »).
+                    //     2. IfAbruptCloseIterator(mappedValue, iteratorRecord).
+                    if_abrupt_close_iterator!(mapfn.call(
+                        this_arg,
+                        &[next.value, k.into()],
+                        context
+                    ))
+                } else {
+                    // vii. Else, let mappedValue be nextValue.
+                    next.value
+                };
+
+                // viii. Let defineStatus be CreateDataPropertyOrThrow(A, Pk, mappedValue).
+                // ix. IfAbruptCloseIterator(defineStatus, iteratorRecord).
+                if_abrupt_close_iterator!(a.create_data_property_or_throw(
+                    k,
+                    mapped_value,
+                    context
+                ));
+            }
+
+            // NOTE(nrabulinski): The loop above has to return before it reaches iteration limit,
```

Comment:
I don't think it's relevant for the documentation who wrote the note :P

```suggestion
            // NOTE: The loop above has to return before it reaches iteration limit,
```